addons/account_analytic_default/models/account_analytic_default.py

```diff
@@ -63,23 +68,29 @@ def account_get(self, product_id=None, partner_id=None, user_id=None, date=None,
 class AccountInvoiceLine(models.Model):
     _inherit = "account.invoice.line"
 
-    @api.onchange('product_id')
-    def _onchange_product_id(self):
-        res = super(AccountInvoiceLine, self)._onchange_product_id()
-        rec = self.env['account.analytic.default'].account_get(self.product_id.id, self.invoice_id.commercial_partner_id.id, self.env.uid,
-                                                               fields.Date.today(), company_id=self.company_id.id)
+    @api.onchange('product_id', 'account_id')
+    def _onchange_product_id_account_id(self):
+        rec = self.env['account.analytic.default'].account_get(product_id=self.product_id.id, partner_id=self.invoice_id.commercial_partner_id.id, account_id=self.account_id.id, user_id=self.env.uid, date=self.invoice_id.date_due, company_id=self.company_id.id)
         self.account_analytic_id = rec.analytic_id.id
         self.analytic_tag_ids = rec.analytic_tag_ids.ids
-        return res
 
-    def _set_additional_fields(self, invoice):
+    def _set_additional_fields(self):
         if not self.account_analytic_id or not self.analytic_tag_ids:
-            rec = self.env['account.analytic.default'].account_get(
-                self.product_id.id, self.invoice_id.commercial_partner_id.id, self.env.uid,
-                fields.Date.today(), company_id=invoice.company_id.id)
+            rec = self.env['account.analytic.default'].account_get(product_id=self.product_id.id, partner_id=self.invoice_id.commercial_partner_id.id, account_id=self.account_id.id, user_id=self.env.uid, date=self.invoice_id.date_due, company_id=self.company_id.id)
```

Comment:
self.invoice_id.company_id instead of self.company_id, because company_id field is related and will be set only on saved invoice lines.